### PR TITLE
Compilation fixes

### PR DESCRIPTION
**Pull request by anstaf**
- fix nvcc11.5 compilation
- fix clang12 compilation with c++20

### CODE REVIEW

File: tests/regression/fn/fn_hd.cpp [hunk 1]
```diff
@@ -33,20 +33,21 @@ constexpr auto rdif = [](auto const &in) { return minus(deref(shift<D, 1>(in)),
...
-};
+constexpr auto flax = [](auto const &in, auto const &lap) { return lambda<flax_helper>(rdif<D>(in), rdif<D>(lap)); };
 
 template <class Param>
-constexpr auto hd = [](auto const &coeff, auto const &in) {
-    return lambda<[](auto const &coeff, auto const &in, auto const &lap) {
-        return minus(deref(in),
-            multiplies(deref(coeff), plus(ldif<i>(Param::flax_i(in, lap)), ldif<j>(Param::flax_j(in, lap)))));
-    }>(coeff, in, Param::lap(in));
+constexpr auto hd_helper = [](auto const &coeff, auto const &in, auto const &lap) {
```
Comment (havogt) [inline tests/regression/fn/fn_hd.cpp:42]: What are these changes about? Clang doesn't understand the pattern?

**Review by havogt — APPROVED** [review]
Just a question.

File: tests/regression/fn/fn_hd.cpp (hunk 1 above)
Comment (anstaf) [reply tests/regression/fn/fn_hd.cpp:42]: Exactly.
`clang12` emits an error on the code like:
```c++
template <auto> struct foo {};
using bar = foo<[]{}>; 
```
Namely it says: `error: a lambda expression cannot appear in this context`
`clang13` is OK with that, as well as  `gcc11`

**Comment by anstaf** [comment]
launch jenkins

**Pull request merged**